### PR TITLE
Fix annoying bugs in nf-core modules lint

### PR DESCRIPTION
**Pull request by drpatelh**
See https://github.com/nf-core/modules/pull/1191#issue-1088039929

### CODE REVIEW

File: nf_core/modules/lint/main_nf.py [hunk 1]
```diff
@@ -163,9 +163,9 @@ def check_process_section(self, lines):
         if re.search("bioconda::", l):
             bioconda_packages = [b for b in l.split() if "bioconda::" in b]
         if re.search("org/singularity", l):
-            singularity_tag = l.split("/")[-1].replace('"', "").replace("'", "").split("--")[-1].strip()
+            singularity_tag = l.split()[0].split("/")[-1].split("/")[-1].replace('"', "").replace("'", "").split(':')[-1]
```
Comment (ewels) [inline nf_core/modules/lint/main_nf.py:166]: Any chance I can replace all of those splits with a regular expression? It makes me ￼a little nervous that it looks quite fragile (but probably unfounded)

What is it trying to do? A code comment with an example string would help for readability too..

Comment (drpatelh) [reply nf_core/modules/lint/main_nf.py:166]: Absolutely! Sorry, I had to move on to the next thing.

Comment (drpatelh) [reply nf_core/modules/lint/main_nf.py:166]: It should be getting the version and build number from the container:

```python
>>> l = "'https://depot.galaxyproject.org/singularity/fastqc:0.11.9--0' :"
>>> singularity_tag = l.split()[0].split("/")[-1].split("/")[-1].replace('"', "").replace("'", "").split(':')[-1]
>>> singularity_tag
'0.11.9--0'

>>> l = "'quay.io/biocontainers/fastqc:0.11.9--0' }\""
>>> docker_tag = l.split()[0].split("/")[-1].split("/")[-1].replace('"', "").replace("'", "").split(':')[-1]
>>> docker_tag
'0.11.9--0'

File: nf_core/modules/lint/main_nf.py [hunk 2]
```diff
@@ -160,12 +167,23 @@ def check_process_section(self, lines):
...
             bioconda_packages = [b for b in l.split() if "bioconda::" in b]
-        if re.search("org/singularity", l):
-            singularity_tag = l.split("/")[-1].replace('"', "").replace("'", "").split("--")[-1].strip()
-        if re.search("biocontainers", l):
-            docker_tag = l.split("/")[-1].replace('"', "").replace("'", "").split("--")[-1].strip()
+        if l.startswith("https://containers") or l.startswith("https://depot"):
+            lspl = l.lstrip("https://").split(":")
+            if len(lspl) == 2:
+                ## e.g. 'https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :
+                singularity_tag = "_".join(lspl[0].split("/")[-1].strip().rstrip(".img").split("_")[1:])
```
Comment (ewels) [inline nf_core/modules/lint/main_nf.py:179]: This stuff has regex written all over it (in my mind). I'll try to refactor when I get a moment, but equally if it works as it is then I'm happy for it to be merged and not to wait.

File: nf_core/modules/lint/main_nf.py [hunk 3]
```diff
@@ -203,16 +221,16 @@ def check_process_section(self, lines):
...
-        line = line.split(",")
-
-        for elem in line:
-            elem = elem.split("(")[1]
-            elem = elem.replace(")", "").strip()
-            input.append(elem)
+        for idx, elem in enumerate(line.split(")")):
+            if elem:
+                elem = elem.split("(")[1]
+                elem = elem.split(",")[0].strip()
```
Comment (ewels) [inline nf_core/modules/lint/main_nf.py:232]: I suspect regex here again would be nicer. Would be nice if you could pop in an example comment as you did above, as it's not clear to me from the code what is going on here really..

**Review by ewels — APPROVED** [review]
Regexes are nice, but otherwise LGTM and happy for a merge 😊

File: nf_core/modules/lint/main_nf.py (hunk 3 above)
Comment (drpatelh) [reply nf_core/modules/lint/main_nf.py:234]: Done in https://github.com/nf-core/tools/pull/1374/commits/0bfa000e0bd0f8f2502c429ca6b8c064b9b0d0cb

File: nf_core/modules/lint/main_nf.py (hunk 2 above)
Comment (drpatelh) [reply nf_core/modules/lint/main_nf.py:179]: I agree. Whenever you have time. It was tricky enough writing the Python code but a regex would be better.

**Pull request merged**